### PR TITLE
DOMA-1934 editable section name

### PR DESCRIPTION
**Pull request by sitozzz**
- [x] Added field for manual set section name
- [x] Display section prefix prefix 

image

image

### CODE REVIEW

**Review by AntonAL — COMMENTED** [review]
When trying to edit section name, I'm getting and error:
> RangeError: Maximum call stack size exceeded
on line 955:
```
const setNameValue = useCallback((value) => setNameValue(Math.abs(value).toString()), [])
```

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx [hunk 1]
```diff
@@ -914,27 +930,56 @@ const UnitForm: React.FC<IUnitFormProps> = ({ builder, refresh }) => {
...
 }
 const MODAL_FORM_EDIT_GUTTER: RowProps['gutter'] = [0, 32]
 const MODAL_FORM_BUTTON_GUTTER: RowProps['gutter'] = [0, 16]
 
-const RemoveSectionForm: React.FC<IRemoveSectionFormProps> = ({ builder, refresh }) => {
+const EditSectionForm: React.FC<IEditSectionFormProps> = ({ builder, refresh }) => {
     const intl = useIntl()
+    const NameLabel = intl.formatMessage({ id: 'pages.condo.property.section.form.name' })
+    const NamePlaceholderLabel = intl.formatMessage({ id: 'pages.condo.property.section.form.name.placeholder' })
+    const SaveLabel = intl.formatMessage({ id: 'Add' })
```
Comment (AntonAL) [inline apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx:944]:
```suggestion
    const SaveLabel = intl.formatMessage({ id: 'Save' })
```

**Comment by sitozzz** [comment]
> When trying to edit section name, I'm getting and error:
> 
> > RangeError: Maximum call stack size exceeded
> > on line 955:
> 
> ```
> const setNameValue = useCallback((value) => setNameValue(Math.abs(value).toString()), [])
> ```

That's a IDE bug on refactor function name

File: apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx (hunk 1 above)
Comment (sitozzz) [reply apps/condo/domains/property/components/panels/Builder/BuildingPanelEdit.tsx]: Fixed

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts [hunk 2]
```diff
@@ -16,7 +16,8 @@ const sectionName = () => {
 const createBuildingMap = (sections: number, sectionTemplate = testSection): MapEdit => {
     const PropertyMap = new MapEdit(null, () => null )
     for (let i = 0; i < sections; i++){
```
Comment (dkoviazin) [inline apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:18]:
```js
const [, ...range] = Array(10).keys()

// [1, 2, 3, 4, 5, 6, 7, 8, 9]

```

File: apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts [hunk 3]
```diff
@@ -16,7 +16,8 @@ const sectionName = () => {
...
-        PropertyMap.addSection({ ...sectionTemplate, name: sectionName(), type: BuildingMapEntityType.Section })
+        const name = (i + 1).toString()
```
Comment (dkoviazin) [inline apps/condo/domains/property/components/panels/Builder/MapConstructor.spec.ts:19]:
```js
const [, ...range] = Array(10).keys()
 // [1, 2, 3, 4, 5, 6, 7, 8, 9]
```

**Review by AntonAL — COMMENTED** [review]
After I hit "Save" button on edit section panel, I see create section panel.
I naturally expect to see a map with all panels closed.

**Comment by sitozzz** [comment]
> After I hit "Save" button on edit section panel, I see create section panel. I naturally expect to see a map with all panels closed.

Done this

**Review by dkoviazin — APPROVED** [review]

**Comment by AntonAL** [comment]
Could you please displace the panel below "Add" button (that is located upper-right corner)?
After clicking on close button on edit panel, a click get's accepted by "Add" button and a dropdown panel appears.
add-button-intercepts-edit-panel-click-event-on-close-button

**Comment by AntonAL** [comment]
> After clicking on close button on edit panel, a click get's accepted by "Add" button and a dropdown panel appears.
> Could you please either displace the edit panel below "Add" button or stop intercepting click event?
> add-button-intercepts-edit-panel-click-event-on-close-button

Added https://doma.atlassian.net/browse/DOMA-1970 to address this issue.

**Review by AntonAL — APPROVED** [review]

**Pull request merged**